src/planner/binder/expression/bind_comparison_expression.cpp

```diff
@@ -125,6 +125,14 @@ BindResult ExpressionBinder::BindExpression(ComparisonExpression &expr, idx_t de
 	// add casts (if necessary)
 	left.expr = BoundCastExpression::AddCastToType(move(left.expr), input_type);
 	right.expr = BoundCastExpression::AddCastToType(move(right.expr), input_type);
+	if (where_clause) {
+		if (left.expr->type == ExpressionType::OPERATOR_CAST && left.expr->return_type.id() == LogicalTypeId::ENUM) {
+			((BoundCastExpression *)left.expr.get())->try_cast = true;
+		}
+		if (right.expr->type == ExpressionType::OPERATOR_CAST && right.expr->return_type.id() == LogicalTypeId::ENUM) {
```

Comment:
Should we have different behavior if the user inserts a cast, vs when we do an implicit cast? Currently the two are treated the same:

```sql
-- implicit cast
SELECT * FROM tbl WHERE ecol=e2col
-- explicit cast
SELECT * FROM tbl WHERE ecol::ENUM2=e2col
```